framework/src/parser/Parser.C

```diff
@@ -1255,7 +1255,7 @@ Parser::extractParams(const std::string & prefix, InputParameters & p)
     }
     else
     {
-      if (p.isPrivate(it.first))
+      if (p.isPrivate(it.first) && !in_global)
```

Comment:
if it s private and in global, should we return here to avoid setting it ?

Reply:
Been too long that I don't have a strong opinion. I'll default to your judgement. Other issues taken care of.

Reply:
> If these changes apply the global parameter to the suppressed parameter this is problematic.
> I think we ll want to check that with a test as well.
Basically if this is happening, which we dont want, then I think you need to return here to make it stop from happening

Reply:
this being "the global parameter ends up being applied to the suppressed parameter"

Reply:
Fixed now. It was indeed being applied